### PR TITLE
Adding gui related fields to config

### PR DESCRIPTION
**Pull request by findolor**
## Motivation

See issue: https://github.com/rainlanguage/rain.orderbook/issues/964

This PR adds the gui field to config for future usage

## Solution

- Add new field `gui` to config source and config

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

### CODE REVIEW

File: crates/settings/src/gui.rs [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    pub value: GuiFieldValueSource,
+}
+
+#[typeshare]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "kebab-case")]
+pub struct GuiDepositSource {
+    pub token: TokenRef,
+    #[typeshare(typescript(type = "number"))]
+    pub min: Option<f64>,
```
Comment (hardyjosh) [inline crates/settings/src/gui.rs:52]: let's remove

Comment (hardyjosh) [reply crates/settings/src/gui.rs:52]: we were going to use json schema for validation

File: crates/settings/src/gui.rs [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+}
+
+#[typeshare]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub struct GuiFieldDefinition {
+    binding: String,
+    name: String,
+    description: String,
+    #[typeshare(typescript(type = "string"))]
+    min: Option<U256>,
```
Comment (hardyjosh) [inline crates/settings/src/gui.rs:244]: remove in favour of json schema validation later

File: crates/settings/src/gui.rs [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+
+#[typeshare]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "kebab-case")]
+pub struct GuiFieldDefinitionSource {
+    pub binding: String,
+    pub name: String,
+    pub description: String,
+    #[typeshare(typescript(type = "number"))]
+    pub min: Option<f64>,
```
Comment (hardyjosh) [inline crates/settings/src/gui.rs:64]: remove

File: crates/settings/src/gui.rs [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    U256,
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, sync::Arc};
+use thiserror::Error;
+use typeshare::typeshare;
+
+use crate::{Deployment, DeploymentRef, Token, TokenRef};
+
+pub const GUI_PRESET_VALUE_DECIMALS: u8 = 18;
```
Comment (hardyjosh) [inline crates/settings/src/gui.rs:13]: Do we need this anymore?

**Review by hardyjosh — CHANGES_REQUESTED** [review]

File: crates/settings/src/gui.rs (hunk 4 above)
Comment (findolor) [reply crates/settings/src/gui.rs:13]: No need for this, removing

**Review by hardyjosh — APPROVED** [review]

**Pull request merged**